### PR TITLE
Mvvm

### PR DESCRIPTION
**Pull request by AldrinBahle**
Hi please review my MVVM structure setup.

### CODE REVIEW

**Review by Sashen943 — APPROVED** [review]
Nice work Aldrin!

File: Mars Mission/Mars Mission.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,16 +7,17 @@
 	objects = {
```
Comment (Sashen943) [inline Mars Mission/Mars Mission.xcodeproj/project.pbxproj:8]: Please provide screenshots for these changes

File: Mars Mission/Mars Mission/APILayer/APILayer.swift [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+//  APILayer.swift
+//  Mars Mission
+//
+//  Created by Aldrin Bahle Gama on 2020/11/04.
+//
+
+import Foundation
+
+struct API {
+    static let nasaAPI = "NYVu6FVU9WwiBBy9zp4iaWveE1SmWlnDFgEy32cH"
```
Comment (Sashen943) [inline Mars Mission/Mars Mission/APILayer/APILayer.swift]: Consider putting these URL end points within a plist

File: Mars Mission/Mars Mission/APILayer/APILayer.swift [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    static let key = nasaAPI
+    static let baseURL = URL(string: "https://api.nasa.gov")!
```
Comment (Sashen943) [inline Mars Mission/Mars Mission/APILayer/APILayer.swift]: Don't forcefully unwrap here

File: Mars Mission/Mars Mission/Model/DetailsModel/WeatherDetails.swift [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+//  WeatherDetails.swift
+//  Mars Mission
+//
+//  Created by Aldrin Bahle Gama on 2020/11/04.
+//
+
+import Foundation
+
+enum WeatherDetails: Int {
+    case sunrise
```
Comment (Sashen943) [inline Mars Mission/Mars Mission/Model/DetailsModel/WeatherDetails.swift:11]: Fix indentation here

File: Mars Mission/Mars Mission/Model/WeatherModel.swift [hunk 5]
```diff
@@ -0,0 +1,35 @@
+//
+//  WeatherModel.swift
+//  Mars Mission
+//
+//  Created by Aldrin Bahle Gama on 2020/11/04.
+//
+
+import Foundation
+
+struct WeatherDataModel: Codable {
```
Comment (Sashen943) [inline Mars Mission/Mars Mission/Model/WeatherModel.swift:10]: Make the variables in the structs optional and use them safely

File: Mars Mission/Mars Mission/APILayer/APILayer.swift (hunk 2 above)
Comment (AldrinBahle) [reply Mars Mission/Mars Mission/APILayer/APILayer.swift]: I've resolved this one and the screenshots of the changes are at the top.

File: Mars Mission/Mars Mission/Model/DetailsModel/WeatherDetails.swift (hunk 4 above)
Comment (AldrinBahle) [reply Mars Mission/Mars Mission/Model/DetailsModel/WeatherDetails.swift:11]: Screenshot 2020-11-05 at 05 37 43

fixed thank you.

File: Mars Mission/Mars Mission/Model/WeatherModel.swift (hunk 5 above)
Comment (AldrinBahle) [reply Mars Mission/Mars Mission/Model/WeatherModel.swift:10]: Screenshot 2020-11-05 at 05 41 12

fixed.

**Review by Sashen943 — APPROVED** [review]
Nice work

**Pull request merged**